### PR TITLE
[MRG] Ica indexing

### PR DESCRIPTION
**Pull request by jaeilepp**
Take over from https://github.com/mne-tools/mne-python/pull/3565 thanks @ahoejlund.

### CODE REVIEW

File: mne/viz/tests/test_ica.py [hunk 1]
```diff
@@ -163,6 +163,12 @@ def test_plot_ica_sources():
                            ecg=False, eog=False, exclude='bads')
     ica = ICA(n_components=2, max_pca_components=3, n_pca_components=3)
     ica.fit(raw, picks=ica_picks)
+    ica.exclude = [1]
+    fig = ica.plot_sources(raw)
+    fig.canvas.key_press_event('escape')
+    # Sadly close_event isn't called on Agg backend and the test always passes.
+    assert_array_equal(ica.exclude, [1])
```
Comment (jaeilepp) [inline mne/viz/tests/test_ica.py]: I couldn't find a reasonable way to test the interactive selection of components on 'Agg' backend, but this should catch the error on other backends.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -1747,3 +1749,5 @@ of commits):
 .. _Chris Mullins: http://crmullins.com
 
 .. _Phillip Alday: http://palday.bitbucket.org
+
+.. _Andreas Hojlund: https://github.com/ahoejlund
```
Comment (jaeilepp) [inline doc/whats_new.rst]: @ahoejlund I added your github page here. Feel free to update it.

Comment (ahoejlund) [inline doc/whats_new.rst]: @jaeilepp Thx heaps.

File: mne/viz/ica.py [hunk 3]
```diff
@@ -400,7 +400,7 @@ def _plot_ica_sources_evoked(evoked, picks, exclude, title, show, labels=None):
     exclude_labels = list()
     for ii in picks:
         if ii in exclude:
-            line_label = 'ICA %03d' % (ii + 1)
+            line_label = 'IC #%03d' % (ii)
```
Comment (agramfort) [inline mne/viz/ica.py]: (ii) -> ii

**Comment by agramfort** [comment]
@ahoejlund does that do the job for you?

File: mne/viz/ica.py (hunk 3 above)
Comment (ahoejlund) [inline mne/viz/ica.py]: thx, just fixed it in #3565 - can you reference that here, @jaeilepp? I'm not sure how or whether I can do that here.

**Comment by ahoejlund** [comment]
@agramfort it most certainly does, thx!

**Comment by agramfort** [comment]
merge when CIs are happy

File: mne/viz/ica.py [hunk 4]
```diff
@@ -842,8 +842,7 @@ def _pick_bads(event, params):
 def _close_event(events, params):
     """Function for excluding the selected components on close."""
     info = params['info']
-    c_names = ['IC #%03d' % (x) for x
-               in range(params['ica'].n_components_)]
+    c_names = ['IC #%03d' % x for x in range(params['ica'].n_components_)]
```
Comment (ahoejlund) [inline mne/viz/ica.py]: (x) -> x
in line 856 below, as well

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**